langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -833,6 +833,27 @@ function testCloneWithTypeWithInferredArgument() {
    assert(n2[1], "world");
 }
 
+type Map map<anydata>;
+
+function testCloneWithTypeWithImmutableTypes() {
+
+   map<int> & readonly m = {a: 1, b: 2};
+   map<map<int>> n = {m};
+   var o = checkpanic n.cloneWithType(Map);
+   assert(o["m"] === m, false);
+   assert(o["m"].isReadOnly(), false);
+   anydata p = o["m"];
+   assert(p is map<anydata>, true);
+   map<anydata> q = <map<anydata>>p;
+   q["c"] = "foo";
```

Comment:
Shouldn't we also assert the value of `q["c"]` and maybe the length of `q` and see?

Reply:
This will panic normally. BTW I'll add a simple check.